navigation/AppNavigation.js

```diff
@@ -1,13 +1,35 @@
+import React from 'react';
 import { createStackNavigator, createAppContainer } from 'react-navigation';
 
 import LibraryContainer from '../app/screens/Library/index';
 import BookDetailContainer from '../app/screens/BookDetail/index';
 
+import ImageHeader from './components/ImageHeader';
+import BackButtonHeader from './components/BackButtonHeader';
+import { styles } from './styles';
+
 const AppNavigator = createStackNavigator({
-  library: { screen: LibraryContainer },
-  bookDetail: { screen: BookDetailContainer }
-}, {
-  initialRouteName: 'library'
-});
+    library: {
+      screen: LibraryContainer,
+      navigationOptions: {
+        title: 'Library'
+      }
+    },
+    bookDetail: {
+      screen: BookDetailContainer,
+      navigationOptions: {
+        title: 'Book Detail'
+      }
+    }
+  },
+  {
+    initialRouteName: 'library',
+    defaultNavigationOptions: {
+      headerBackground: <ImageHeader />,
```

Comment:
for the custom header component I think it'd be better to use `headerTitle` instead. Also I don't think you need to use jsx for this. Try `headerTitle: ImageHeader` and see if it works

Reply:
i didn't find the way to solve this. `headerBackground` wait a component to render, if i pass a function this prop will fail.